lib/internal/modules/helpers.js

```diff
@@ -128,13 +128,13 @@ const urlToFileCache = new SafeMap();
 function makeRequireFunction(mod, redirects) {
   // lazy due to cycle
   const Module = lazyModule();
-  if (mod instanceof Module !== true) {
-    throw new ERR_INVALID_ARG_TYPE('mod', 'Module', mod);
-  }
 
   /** @type {RequireFunction} */
   let require;
   if (redirects) {
+    if (mod instanceof Module !== true) {
+      throw new ERR_INVALID_ARG_TYPE('mod', 'Module', mod);
+    }
```

Comment:
This doesn’t seem correct; there’s another reference to `mod` in the `else` branch farther down; why would we validate the input only for this branch and not that one?

Reply:
This was added specifically to solve https://github.com/nodejs/node/commit/15bced0bde. Since this check didn't exist previously I think it's _safe_ to ignore the other branches.

Reply:
Well at the very least it looks like a mistake; if I were refactoring this, I would instinctively move the check to the top of the function. There should be a comment explaining why `mod` is always defined for `if (redirects)` but only sometimes defined otherwise.

Reply:
What's missing is a test that would fail if this check was at the top of the function.